src/module_2/module_2.md

```diff
@@ -0,0 +1,3512 @@
+
+
+```python
+orders_with_inventory['order_date'] = pd.to_datetime(orders_with_inventory['order_date'])
+
+weekly_avg_ordered_items  = orders_with_inventory.groupby(['order_date', 'id'])['ordered_items'].count().reset_index()
+
+weekly_avg_ordered_items = weekly_avg_ordered_items.groupby(weekly_avg_ordered_items['order_date'].dt.to_period("W"))['ordered_items'].mean()
+plt.figure(figsize=(8, 4))
+weekly_avg_ordered_items.plot(kind='line', marker='.', color='orange')
+plt.xlabel('Year-Week')
+plt.ylabel('Time')
+plt.title('Avg items per basket over time')
+
+
+```
+
+
+
+
+    Text(0.5, 1.0, 'Avg items per basket over time')
+
+
+
+
+    
+![png](module_2_files/module_2_54_1.png)
+    
+
+
+We see a decrease in number of items per basket. We could argue that people have decided to buy less items, possibly motivated by change of inventory from the store. The store might have reduced the total assortment simplifying the total items on sale. This is a typical pattern in some retailers, where they perform assortment optimization after researching the optimal numbers of EANs in their portfolio. Let's see if that has been materialized in the total value $ over time.
+
+
+```python
+value_time = orders_with_inventory.groupby(orders_with_inventory['order_date'].dt.to_period("W"))[['price']].sum()
+
+plt.figure(figsize=(10, 6))
+value_time.plot(kind='line', marker=".", color='orange')
+plt.xlabel('Time')
+plt.ylabel('£')
+plt.title('Volume (£) over time')
+
+#Fig size does not adapt here idk why
+
+```
+
+
+
+
+    Text(0.5, 1.0, 'Volume (£) over time')
+
+
+
+
+    <Figure size 1000x600 with 0 Axes>
+
+
+
+    
+![png](module_2_files/module_2_56_2.png)
+    
+
+
+Volume value indeed has been increasing since beginning. We see a peak Jan-Feb'22 that could be interpreted (event, promo, discount, etc), and then a sudden crease in the last weeks until today - most probably bc data was not yet collected for that last one. This graphic is, again, very similar to orders and avg items over time as it is highly correlated. Let's take a look at the avg basket value over time finally:
+
+
+```python
+avg_value = orders_with_inventory.dropna(subset='price').groupby(['id', 'order_date'])['price'].sum().reset_index()
+avg_value = avg_value.groupby(avg_value['order_date'].dt.to_period("W"))['price'].mean()
+
+plt.figure(figsize=(10, 6))
+avg_value.plot(kind='line', marker=".", color='orange')
+plt.xlabel('Time')
+plt.ylabel('£')
+plt.title('Avg basket value over time')
+
+```
+
+
+
+
+    Text(0.5, 1.0, 'Avg basket value over time')
+
+
+
+
+    
+![png](module_2_files/module_2_58_1.png)
+    
+
+
+
+```python
+avg_value = orders_with_inventory.dropna(subset='price').groupby(['id', 'order_date'])['price'].sum().reset_index()
+avg_value[avg_value['order_date'] > '2020-01-01']['price'].mean()
+```
+
+
+
+
+    56.13289594315838
+
+
+
+So the value has been steadily increasing until today, where it has been more or less stabilized around £55
```

Comment:
I think at this point you can conclude this section with 4 bullet points:
* The change on the total volume (revenue)
* How this change is motivated by the number of orders
* How this change is motivated by the average basket price
* How this change is motivated by the number of items in the basket